src/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -39,5 +42,37 @@ public void SendAsync_SimpleGet_Success()
                 _output.WriteLine(responseContent);
             }
         }
+
+        [Fact]
+        [OuterLoop]
+        public void SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
+        {
+            var handler = new WinHttpHandler();
+            using (var client = new HttpClient(handler))
+            {
+                var cts = new CancellationTokenSource();
+                Task<HttpResponseMessage> t = client.GetAsync(SlowServer, cts.Token);
+
+                Thread.Sleep(500);
```

Comment:
This particular test is not async.  It returns void.  Can I even use Task.Delay here?

What is the advantage of it over Thread.Sleep especially in this test which is synchronous.

@stephentoub Can you comment on this?  Thx.